freqtrade/strategy/interface.py

```diff
@@ -112,10 +131,24 @@ def analyze_ticker(self, ticker_history: List[Dict], metadata: dict) -> DataFram
         add several TA indicators and buy signal to it
         :return DataFrame with ticker data and indicator data
         """
+
+        # Get last candle processed and ln of Dataframe
+        last_candle_processed = self.r.get_candle_row()
         dataframe = parse_ticker_dataframe(ticker_history)
-        dataframe = self.advise_indicators(dataframe, metadata)
-        dataframe = self.advise_buy(dataframe, metadata)
-        dataframe = self.advise_sell(dataframe, metadata)
+
+        if last_candle_processed != len(dataframe.index):
+            # Defs that only make change on new candle data here
+            dataframe = self.advise_indicators(dataframe, metadata)
+            dataframe = self.advise_buy(dataframe, metadata)
+            dataframe = self.advise_sell(dataframe, metadata)
+            self.r.set_candle_row(len(dataframe.index))
```

Comment:
I'm not sure this works as it shoulld - the amount (length) of candles is ccxt default (i guess 500) apart from backtesting - but shifting in time.

you should rather gather the time of the last candle analyzed `dataframe.iloc[-1]['date']` (or similar - not sure if syntax is 100% correct).

Reply:
thats a good call - thanks